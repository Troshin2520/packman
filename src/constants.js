export const BLOCK_SIZE = 4;

export const ACTION_CHANGE_PACMAN_DIRECTION = 'CHANGE_PACMAN_DIRECTION';
export const ACTION_CHANGE_ZONE = 'CHANGE_ZONE';
export const ACTION_MOVE_SPIDER = 'MOVE_SPIDER';
export const ACTION_MOVE_PACMAN = 'MOVE_PACMAN';

export const zones = [
  {
    field: [
      [4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4],
      [4, 2, 1, 1, 1, 1, 4, 1, 1, 1, 1, 2, 4],
      [4, 1, 4, 4, 4, 1, 4, 1, 4, 4, 4, 1, 4],
      [4, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 4],
      [4, 1, 4, 0, 0, 0, 0, 0, 0, 0, 4, 1, 4],
      [4, 1, 4, 0, 4, 8, 8, 8, 4, 0, 4, 1, 4],
      [0, 1, 4, 0, 4, 0, 0, 0, 4, 0, 4, 1, 0],
      [4, 1, 4, 0, 4, 4, 4, 4, 4, 0, 4, 1, 4],
      [4, 1, 4, 1, 1, 1, 0, 1, 1, 1, 4, 1, 4],
      [4, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 4],
      [4, 1, 4, 4, 4, 1, 4, 1, 4, 4, 4, 1, 4],
      [4, 2, 1, 1, 1, 1, 4, 1, 1, 1, 1, 2, 4],
      [4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4]
    ],
    speed: 0.6,
    pacman: {x: 6, y: 8, move: 'no', next: 'no'},
    spiders: {
      red:{x: 6, y: 6, move: 'up'},
      green:{x: 6, y: 6, move: 'up'},
      orange:{x: 5, y: 6, move: 'up'},
      blue:{x: 7, y: 6, move: 'up'}
    },
  },
  {
    field: [
      [4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4],
      [4, 2, 1, 1, 1, 1, 4, 1, 1, 1, 1, 4, 4],
      [4, 1, 4, 4, 4, 1, 4, 1, 4, 4, 4, 1, 4],
      [4, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 4],
      [4, 1, 4, 1, 0, 0, 0, 0, 0, 1, 4, 1, 4],
      [4, 1, 4, 1, 4, 8, 8, 8, 4, 1, 4, 1, 4],
      [0, 1, 4, 1, 4, 0, 0, 0, 4, 1, 4, 1, 0],
      [4, 1, 4, 1, 4, 4, 4, 4, 4, 1, 4, 1, 4],
      [4, 1, 4, 1, 1, 1, 1, 1, 1, 1, 4, 1, 4],
      [4, 1, 4, 1, 4, 1, 4, 1, 4, 1, 4, 1, 4],
      [4, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 4],
      [4, 1, 4, 4, 4, 1, 4, 1, 4, 4, 4, 1, 4],
      [4, 2, 1, 1, 1, 1, 4, 1, 1, 1, 1, 2, 4],
      [4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4]
    ],
    speed: 0.5,
    pacman: {x: 6, y: 8, move: 'no', next: 'no'},
    spiders: {
      red:{x: 6, y: 6, move: 'up'},
      green:{x: 6, y: 6, move: 'right'},
      orange:{x: 5, y: 6, move: 'left'},
      blue:{x: 7, y: 6, move: 'down'}
    }
  },
  {
    field: [
      [4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4],
      [4, 2, 1, 1, 1, 1, 4, 1, 1, 1, 1, 2, 4],
      [4, 1, 4, 1, 4, 1, 4, 1, 4, 1, 4, 1, 4],
      [4, 1, 4, 4, 4, 1, 4, 1, 4, 4, 4, 1, 4],
      [4, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 4],
      [4, 1, 4, 1, 0, 0, 0, 0, 0, 1, 4, 1, 4],
      [4, 1, 4, 1, 4, 8, 8, 8, 4, 1, 4, 1, 4],
      [0, 1, 4, 1, 4, 0, 0, 0, 4, 1, 4, 1, 0],
      [4, 1, 4, 1, 4, 4, 4, 4, 4, 1, 4, 1, 4],
      [4, 1, 4, 1, 1, 1, 1, 1, 1, 1, 4, 1, 4],
      [4, 1, 4, 1, 4, 4, 4, 4, 4, 1, 4, 1, 4],
      [4, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 4],
      [4, 1, 4, 1, 4, 1, 4, 1, 4, 1, 4, 1, 4],
      [4, 1, 4, 4, 4, 1, 4, 1, 4, 4, 4, 1, 4],
      [4, 2, 1, 1, 1, 1, 4, 1, 1, 1, 1, 2, 4],
      [4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4]
    ],
    speed: 0.4,
    pacman: {x: 6, y: 9, move: 'no', next: 'no'},
    spiders: {
      red:{x: 6, y: 7, move: 'up'},
      green:{x: 6, y: 7, move: 'right'},
      orange:{x: 5, y: 7, move: 'left'},
      blue:{x: 7, y: 7, move: 'down'}
    }
  }
];