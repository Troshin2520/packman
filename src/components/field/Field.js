import React, {Component} from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import {ACTION_FEED_ATE, ACTION_PILL_ATE} from '../../constants';
import Breaks from '../breaks/Breaks';
import Way from '../way/Way';
import Gates from '../gates/Gates';
import './Field.less';

class Field extends Component {

  constructor(props) {
    super(props);
    this.updateWayState = this.updateWayState.bind(this);
  }

  updateWayState(action) {
    this.props.onChangeState(action === 1 ? ACTION_FEED_ATE : ACTION_PILL_ATE, {});
  }

  render() {
    const field = this.props.field.map((line, i) => {
      return (<div
        key={i}
        tabIndex="1"
        className="row">
        {line.map((cell, j) => {
          let key = `w${i}.${j}`;
          switch (cell) {
            case 0:
            case 1:
            case 2:
              return <Way
                x={j}
                y={i}
                key={key}
                has={cell}
                onChangeState={this.updateWayState}
              />
            case 4:
              return <Breaks key={key}/>
            case 8:
              return <Gates key={key}/>
            default:
          }
          return '';
        })}
      </div>)
    });
    return <div className="field">{field}</div>;
  }
}

Field.propTypes = {};


export default connect(
  ({field}) => {
    return {field};
  },
  dispatch => ({
    onChangeState(type, item) {
      dispatch({type: type, payload: item});
    }
  })
)(Field);
