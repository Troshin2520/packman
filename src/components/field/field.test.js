import React from 'react';
import configureStore from 'redux-mock-store';
import Renderer from 'react-test-renderer';
import Field from './Field';
import {directions} from "../../constants";


const initialState = {
  field: [
    [4, 1, 4, 0, 0, 0, 0, 0, 0, 0, 4, 1, 4],
    [4, 1, 4, 0, 4, 8, 8, 8, 4, 0, 4, 1, 4]
  ],
  speed: 0.6,
  pacman: {x: 6, y: 8, dir: directions.left},
  spiders: {
    red: {x: 6, y: 6, move: directions.up}
  },
};

const mockStore = configureStore();
const store = mockStore(initialState);

describe('Gates tests', () => {

  it('Gates renders without crushing', () => {
    const component = Renderer.create(
      <Field store={store}/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
