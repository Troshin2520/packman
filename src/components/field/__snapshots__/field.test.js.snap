// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gates tests Gates renders without crushing 1`] = `
<div
  className="field"
>
  <div
    className="row"
    tabIndex="1"
  >
    <div
      className="wall"
    >
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
    </div>
    <div
      className="way feed"
    >
      <div />
    </div>
    <div
      className="wall"
    >
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
    </div>
    <div
      className="way empty"
    >
      <div />
    </div>
    <div
      className="way empty"
    >
      <div />
    </div>
    <div
      className="way empty"
    >
      <div />
    </div>
    <div
      className="way empty"
    >
      <div />
    </div>
    <div
      className="way empty"
    >
      <div />
    </div>
    <div
      className="way empty"
    >
      <div />
    </div>
    <div
      className="way empty"
    >
      <div />
    </div>
    <div
      className="wall"
    >
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
    </div>
    <div
      className="way feed"
    >
      <div />
    </div>
    <div
      className="wall"
    >
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
    </div>
  </div>
  <div
    className="row"
    tabIndex="1"
  >
    <div
      className="wall"
    >
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
    </div>
    <div
      className="way feed"
    >
      <div />
    </div>
    <div
      className="wall"
    >
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
    </div>
    <div
      className="way empty"
    >
      <div />
    </div>
    <div
      className="wall"
    >
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
    </div>
    <div
      className="gates"
    >
      <div />
    </div>
    <div
      className="gates"
    >
      <div />
    </div>
    <div
      className="gates"
    >
      <div />
    </div>
    <div
      className="wall"
    >
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
    </div>
    <div
      className="way empty"
    >
      <div />
    </div>
    <div
      className="wall"
    >
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
    </div>
    <div
      className="way feed"
    >
      <div />
    </div>
    <div
      className="wall"
    >
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
      <div
        className="stone"
      />
      <div
        className="stone-half"
      />
    </div>
  </div>
</div>
`;
